muni_portal/tests/api/test_user_account.py

```diff
@@ -0,0 +1,73 @@
+from django.contrib.auth.models import User
+from django.urls import reverse
+from faker import Faker
+from rest_framework import status
+from rest_framework.test import APITestCase
+from unittest.mock import patch
+from urllib.parse import urlparse, parse_qsl
+
+
+class ApiUserAccountTestCase(APITestCase):
+
+    def setUp(self):
+        self.password = Faker().password(length=8)
+
+    @patch("rest_registration.notifications.email.send_notification")
+    @patch("rest_registration.notifications.email.create_verification_notification")
+    def test_registration(self, create_verification_notification_mock, send_notification_mock):
+        data = {
+            "email": "test@test.com",
+            "username": "test",
+            "password": self.password,
+            "password_confirm": self.password
+        }
+        response = self.client.post(reverse("rest_registration:register"), data=data)
+        assert response.status_code == status.HTTP_201_CREATED
+        assert send_notification_mock.called
+
+        # TODO: check verification flow
+        verification_url = urlparse(
+            create_verification_notification_mock.call_args[0][3]["params_signer"].get_url()
+        )
+        response = self.client.post(verification_url.path, data=dict(parse_qsl(verification_url.query)))
+        assert response.status_code == status.HTTP_200_OK
+
+        response = self.client.post(reverse("token_obtain_pair"), data=data)
+        assert response.status_code == status.HTTP_200_OK
+
+    @patch("rest_registration.notifications.email.send_notification")
+    @patch("rest_registration.notifications.email.create_verification_notification")
+    def test_reset_password(self, create_verification_notification_mock, send_notification_mock):
+        user = User.objects.create_user(
+            username="test", email="test@test.com", password=self.password
+        )
+        data = {"email": user.email}
+        response = self.client.post(reverse("rest_registration:send-reset-password-link"), data=data)
+        assert response.status_code == status.HTTP_200_OK
+        assert send_notification_mock.called
+
+        # TODO: check verification flow
+        verification_url = urlparse(
+            create_verification_notification_mock.call_args[0][3]["params_signer"].get_url()
+        )
+        reset_password_data = dict(parse_qsl(verification_url.query))
+        reset_password_data["password"] = Faker().password(length=8)
+        response = self.client.post(verification_url.path, data=reset_password_data)
+        assert response.status_code == status.HTTP_200_OK
+
+        reset_password_data["username"] = user.username
+        response = self.client.post(reverse("token_obtain_pair"), data=reset_password_data)
+        assert response.status_code == status.HTTP_200_OK
+
+    def test_profile(self):
```

Comment:
```suggestion
    def test_profile_authorization(self):
```

Please also assert that a user gets denied when requesting another user's profile

Reply:
There is no way to retrieve another user's profile, only if you steal its token because we get user id from auth header.

Reply:
oh ok - so the profile is identified by the token.

Can we expose custom user model fields like this? I'll add that as a separate task.

Reply:
Yes, we can